Code/Tools/TestImpactFramework/Runtime/Common/Code/Source/Artifact/Factory/TestImpactTestEnumerationSuiteFactory.cpp

```diff
@@ -84,4 +87,74 @@ namespace TestImpact
             return testSuites;
         }
     } // namespace GTest
+
+    namespace Python
+    {
+        AZStd::vector<AZStd::pair<AZStd::string, AZStd::vector<TestEnumerationSuite>>> TestEnumerationSuitesFactory(
+            const AZStd::string& testEnumerationData)
+        {
+            const char newLineDelim = '\n';
+            size_t startLine;
+            size_t endLine = 0;
+            int count = 0;
+
+            AZStd::vector<AZStd::pair<AZStd::string, AZStd::vector<TestEnumerationSuite>>> pairs;
+            AZStd::map<AZStd::string, AZStd::vector<TestEnumerationSuite>> testSuiteMap;
+            AZStd::string previousTestFixture;
+            // Steps through our string line by line. Assumes lines are split by newline characters.
+            while ((startLine = testEnumerationData.find_first_not_of(newLineDelim, endLine)) != AZStd::string::npos)
+            {
+                endLine = testEnumerationData.find(newLineDelim, startLine);
+                AZStd::string curLine = testEnumerationData.substr(startLine, endLine - startLine);
+
+                // This regex matches pytest test names, in the form ScriptPath::TestFixture::TestName. It will accept both parameterized
+                // and unparamaterized tests (with or without [] at the end).
+                AZStd::basic_regex testNamePattern = AZStd::basic_regex("(.*py)::([A-Za-z_/\\-0-9]*)::(.*)");
+                AZStd::smatch matchResults;
+                if (AZStd::regex_search(curLine, matchResults, testNamePattern))
+                {
+                    AZStd::string absoluteScriptPath = matchResults[1];
+                    AZStd::string testFixture = matchResults[2];
+                    AZStd::string testName = matchResults[3];
```

Comment:
suggestion: Should we check the size of `matchResults` before subscripting into it?

Reply:
I would suggest an `AZ_TestImpact_Eval` to throw an exception if the number of match results is not equal to the number of capture groups + 1, possibly even replacing those num literals with enum values (starting from `1`).